src/compiler/checker.ts

```diff
@@ -30405,7 +30422,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         // We only look for uninitialized variables in strict null checking mode, and only when we can analyze
         // the entire control flow graph from the variable's declaration (i.e. when the flow container and
         // declaration container are the same).
-        const assumeInitialized = isParameter || isAlias || isOuterVariable || isSpreadDestructuringAssignmentTarget || isModuleExports || isSameScopedBindingElement(node, declaration) ||
+        const assumeInitialized = isParameter || isAlias ||
+            (isOuterVariable && !((immediateDeclaration && isVariableDeclaration(immediateDeclaration) && !immediateDeclaration.initializer && !immediateDeclaration.exclamationToken && isMutableLocalVariableDeclaration(symbol, immediateDeclaration)) && !isSymbolAssignedDefinitely(symbol))) ||

```

Comment:
Haha, that was my initial thought as well, but I'm not keen on the idea of having to perform additional evaluations even when isParameter or isAlias is true, although this may not impact the overall performance.

https://github.com/microsoft/TypeScript/pull/55887/commits/bf4bb51b23d29470930781bf682d52b30e35ea9a

Reply:
I don't think it would matter that much, the checks being performed should be very cheap, but it's up to you.

Reply:
It has been relocated to isNeverInitialized, resulting in cleaner code now.